### PR TITLE
Adding a Dialog Box

### PR DESCRIPTION
**Pull request by RiByryn**
1. Added a dialog box call to the test suite page
2. Added dialog boxes: 
- basic
image
- for entering text
image
- for entering a number
image

3. Added data transfer from the dialog box input field to the API
4. Added handling of closing the dialog box: when the window is closed, a warning is displayed and the tests stop
image
5. Added an example of tests with dialog boxes of all types
6. Added a warning display in case of entering an empty value
image
7. The index.tsx file has been adjusted to work in developer mode

### CODE REVIEW

File: docs/documentation/hardpy_panel.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ The **hardpy panel** or operator panel is a web interface that displays and cont
 **HardPy panel** allows you to:
 
 - Start and stop testing.
+- Interact with the dialog boxes during testing.
```
Comment (xorialexandrov) [inline docs/documentation/hardpy_panel.md:10]: Need to merge with 

```
Browse dialog box (TODO: Add ref).
```

File: examples/project/dialog_box/test_dialog_box.py [hunk 2]
```diff
@@ -10,7 +10,7 @@ def test_base_dialog_box():
         dialog_text="If you are not sleeping, press the Confirm button",
     )
     response = hardpy.run_dialog_box(info)
-    assert response == "ok"
+    assert response == None
```
Comment (xorialexandrov) [inline examples/project/dialog_box/test_dialog_box.py:13]:
```python
assert response is None 
```

File: examples/project/dialog_box/test_dialog_box.py [hunk 3]
```diff
@@ -37,9 +37,9 @@ def test_dialog_box_with_num_input():
...
 
     info = hardpy.dialog_box.DialogBoxData(
-        dialog_text="Enter the number 5 and press the Confirm button",
+        dialog_text="Enter the number 123 and press the Confirm button",
         title_bar="Example of entering a number",
         widget=widget,
     )
     response = hardpy.run_dialog_box(info)
-    assert response == "ok"
+    assert response == 123.0
```
Comment (xorialexandrov) [inline examples/project/dialog_box/test_dialog_box.py:45]: Need convert to `int` and check it.

File: docs/changelog.md [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 Versions follow [Semantic Versioning](https://semver.org/): `<major>.<minor>.<patch>`.
 
+* Add dialog boxes.
```
Comment (xorialexandrov) [inline docs/changelog.md:5]: Need to add the types of dialog boxes supported.

File: docs/documentation/hardpy_panel.md (hunk 1 above)
Comment (RiByryn) [reply docs/documentation/hardpy_panel.md:10]: Ready

File: examples/project/dialog_box/test_dialog_box.py (hunk 2 above)
Comment (RiByryn) [reply examples/project/dialog_box/test_dialog_box.py:13]: Ready

File: examples/project/dialog_box/test_dialog_box.py (hunk 3 above)
Comment (RiByryn) [reply examples/project/dialog_box/test_dialog_box.py:45]: Ready

File: docs/changelog.md (hunk 4 above)
Comment (RiByryn) [reply docs/changelog.md:5]: Ready

File: docs/documentation/hardpy_panel.md [hunk 5]
```diff
@@ -27,6 +27,41 @@ After this open page http://localhost:8000/ in the browser.
...
+
+- **Dialog box with a text input field**
+
+Contains an instruction or question, a text input field, and a `confirm` button for confirmation
+
+![text_input_dialog_box](../img/text_input_dialog_box.jpg)
+
+- **Dialog box with a number input field**
+
+Contains an instruction or question, a text input field, and a `confirm` button for confirmation. Allows entering only numbers with a period separator, allows entering numbers using `e`. The entered numbers will be converted to float
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply docs/documentation/hardpy_panel.md]: Ready

File: docs/documentation/hardpy_panel.md [hunk 6]
```diff
@@ -27,6 +27,41 @@ After this open page http://localhost:8000/ in the browser.
 When the operator panel is running, you can run tests through the web interface or through
 the pytest launcher (in a terminal or from another application).
 
+### Dialog box
+
+For user interaction with the test, it is possible to use dialog boxes.
+
+Currently, there are three types of dialog boxes:
+
+- **Basic dialog box**
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply docs/documentation/hardpy_panel.md]: Ready

File: docs/documentation/hardpy_panel.md [hunk 7]
```diff
@@ -27,6 +27,41 @@ After this open page http://localhost:8000/ in the browser.
...
+![text_input_dialog_box](../img/text_input_dialog_box.jpg)
+
+- **Dialog box with a number input field**
+
+Contains an instruction or question, a text input field, and a `confirm` button for confirmation. Allows entering only numbers with a period separator, allows entering numbers using `e`. The entered numbers will be converted to float
+
+![num_input_dialog_box](../img/num_input_dialog_box.jpg)
+
+
+If the user clicks `confirm` without entering anything, a warning window will be displayed.
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply docs/documentation/hardpy_panel.md:60]: Ready

File: docs/documentation/hardpy_panel.md [hunk 8]
```diff
@@ -27,6 +27,41 @@ After this open page http://localhost:8000/ in the browser.
...
+
+![alert](../img/alert.jpeg)
+
+If the user closes the dialog box (using the cross in the upper right corner), the tests will be stopped, an error message will be displayed.
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply docs/documentation/hardpy_panel.md:66]: Ready

File: docs/documentation/hardpy_panel.md [hunk 9]
```diff
@@ -27,6 +27,41 @@ After this open page http://localhost:8000/ in the browser.
...
+
+![notification](../img/notification.jpeg)
+
+An example of usage can be seen in the example [dialog box](../examples/dialog_box.md)
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply docs/documentation/hardpy_panel.md]: Ready

File: docs/examples/dialog_box.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+```ini
+[pytest]
+log_cli = true
+log_cli_level = INFO
+log_cli_format = %(asctime)s [%(levelname)s] %(message)s
+log_cli_date_format = %H:%M:%S
+addopts = --hardpy-pt
+```
+
+### conftest.py
```
[the remark this answers is not in the document]
Comment (RiByryn) [reply docs/examples/dialog_box.md]: Ready

**Review by xorialexandrov — APPROVED** [review]

**Pull request merged**